### ISSUE TITLE
recipe is incompatible with newer version of python-eccodes

### ISSUE THREAD

**Issue by TomAugspurger** [conda-forge/cfgrib-feedstock#25]
Currently, the recipe requires python-eccodes>=2020.06: https://github.com/conda-forge/cfgrib-feedstock/blob/d48670ee34858cd7b3fd3ee9d8da65ca1e302cc7/recipe/meta.yaml#L28

It seems that in https://github.com/conda-forge/python-eccodes-feedstock/pull/77/files, python-eccodes moved from a date-based versioning like 2021.05.1 to 1.4.0. Sine 2021 > 1, you can't install cfgrib==0.9.9.1 from conda-forge

```python
$ mamba create -n cfgrib-test -c conda-forge "cfgrib>=0.9.9.1"
Encountered problems while solving.
Problem: nothing provides python-eccodes >=2020.06 needed by cfgrib-0.9.9.1-pyhd8ed1ab_1
```

cc @iainrussell from https://github.com/conda-forge/python-eccodes-feedstock/pull/77/ and @alexamici. I don't really know what action is best going forward. I assume that the change in the `python-eccodes` feedstock was made to follow some upstream change, and so conda-forge will just have to adapt. I also don't see the date-based versions of python-eccodes on https://pypi.org/project/eccodes-python/#history. Were they yanked, or just never existed?

We need to change something, so that users can install cfgrib from conda-forge. Maybe an upper bound on the `python-eccodes` dependency, that's lower than the lowest date-based version?

**Comment by iainrussell** [comment]
Hi Tom, you're quite right, thank you for bringing this to our attention! Indeed, I did mark all the date-based python-eccodes in conda as 'broken' so that they would no longer be available. These tags were only there accidentally, and they caused a problem - even though version 2021.05.0 has a larger number than 1.4.0, it is older. So now, the proper versioning has been restored and anything that was relying on those date-based version numbers will have to be updated. I don't think I have admin rights on this feedstock, but I'll put in a pull request to change the dependency.
Cheers,
Iain

**Comment by TomAugspurger** [comment]
Closed by #26. Thanks all!
